### PR TITLE
don't panic if trying to get a job on a non existsing action

### PR DESCRIPTION
**Pull request by zaibon**

### CODE REVIEW

File: JumpScale9AYS/ays/lib/Service.py [hunk 1]
```diff
@@ -769,7 +769,11 @@ def _executeActionService(self, action, args={}):
             return result
 
     def getJob(self, actionName, args={}, context=None):
-        action = self.model.actions[actionName]
+        try:
+            action = self.model.actions[actionName]
+        except KeyError:
+            return j.exceptions.Input("action %s dos not exists on %s" % str(self))
```
Comment (rkhamis) [inline JumpScale9AYS/ays/lib/Service.py]: Shouldn't that be a raise?

Comment (zaibon) [reply JumpScale9AYS/ays/lib/Service.py]: yep it should

Comment (zaibon) [reply JumpScale9AYS/ays/lib/Service.py]: Also, do you think `Input` is correct ?

**Comment by zaibon** [comment]
> Actually, no. We should do it with a if not actionName in self.model.actions... no need to try/except

What's the difference ?

File: JumpScale9AYS/ays/lib/Service.py (hunk 1 above)
Comment (rkhamis) [reply JumpScale9AYS/ays/lib/Service.py]: KeyError is the most descriptive, actually. But input works.

**Comment by rkhamis** [comment]
> What's the difference ?
cleaner if nothing else.

File: JumpScale9AYS/ays/lib/Service.py [hunk 2]
```diff
@@ -762,7 +762,11 @@ def _executeActionService(self, action, args={}):
             return result
 
     def getJob(self, actionName, args={}, context=None):
-        action = self.model.actions[actionName]
+        try:
+            action = self.model.actions[actionName]
+        except KeyError:
+            raise j.exceptions.Input("action %s dos not exists on %s" % str(self))
```
Comment (ahussein) [inline JumpScale9AYS/ays/lib/Service.py]: This message in the raise exception has incorrect string substitution, the code has 2 %s and only one is replaced !

**Pull request merged**